<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>
  <arg name="open_rviz" default="true"/>
  <arg name="open_gazebo" default="true"/>
  <arg name="team_size" value="4"/>
<!--
  <arg name="map_path1" default="$(find turtlebot3_nps)/maps/multi_map.yaml"/>
  <arg name="map_path2" default="$(find turtlebot3_navigation)/maps/map.yaml"/>
  
  <arg name="world_path1" default="$(find turtlebot3_nps)/worlds/watkinslab.world"/>
  <arg name="world_path2" default="$(find turtlebot3_gazebo)/worlds/turtlebot3_world.world"/>
 -->
 <arg name="map_empty" default="$(find multi_amr_nectec)/maps/map.yaml"/>
 <arg name="emptyworld" default="$(find turtlebot3_gazebo)/worlds/empty.world"/>
  <!-- Map server --> 
 <node pkg="map_server" type="map_server" name="map_server" args="$(arg map_empty)" respawn="false" >
    <param name="frame_id" value="/map" />
  </node>
  <!-- start gazebo --> 
<include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="default"/> 
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="$(arg open_gazebo)"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
</include>
 <!-- RVIZ  --> 
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
         args="-d $(find multi_amr_nectec)/rviz/4_amr.rviz"/>
  </group>

<!-- AMR 0  -->
  <group ns="amr_0">
    <include file="$(find multi_amr_nectec)/launch/single_amr.launch" >
      <arg name="init_pose_gazebo" value="-x 3 -y 2 -z 0 -Y 0 -P 0 -R 0" />
      <arg name="init_pose_rviz" value="0 0 0 0 0 0" />    
      <arg name="robot_name" value="amr_0" />
    </include>
     <node pkg="multi_amr_nectec" type="get_odom_robot" name="get_odom_robot" output="screen" />
  </group>

<!-- AMR 1  -->
 <group ns="amr_1">
    <include file="$(find multi_amr_nectec)/launch/single_amr.launch" >
      <arg name="init_pose_gazebo" value="-x -2 -y 4 -z 0 -Y 0 -P 0 -R 0" />  
      <arg name="init_pose_rviz" value="0 0 0 0 0 0" /> 
      <arg name="robot_name" value="amr_1" />
    </include> 
    <node pkg="multi_amr_nectec" type="get_odom_robot" name="get_odom_robot" output="screen" /> 
  </group>

<!-- AMR 2   --> 
 <group ns="amr_2">
    <param name="tf_prefix" value="amr_2" />
    <param name="amcl/initial_pose_x" value="0" />
    <param name="amcl/initial_pose_y" value="0" />
    <include file="$(find multi_amr_nectec)/launch/single_amr.launch" >
      <arg name="init_pose_gazebo" value="-x -1 -y -2 -z 0 -Y 0" />  
      <arg name="init_pose_rviz" value="0 0 0 0 0 0" />     
      <arg name="robot_name" value="amr_2" />
    </include> 
    <node pkg="multi_amr_nectec" type="get_odom_robot" name="get_odom_robot" output="screen" /> 
  </group>

<!-- AMR 3   -->
 <group ns="amr_3">
   <param name="tf_prefix" value="amr_3" />
    <param name="amcl/initial_pose_x" value="0" />
    <param name="amcl/initial_pose_y" value="0" />
    <include file="$(find multi_amr_nectec)/launch/single_amr.launch" >
      <arg name="init_pose_gazebo" value="-x 2 -y -4 -z 0 -Y 0 -P 0 -R 0" />  
      <arg name="init_pose_rviz" value="0 0 0 0 0 0" /> 
      <arg name="robot_name" value="amr_3" />
    </include>
    <node pkg="multi_amr_nectec" type="get_odom_robot" name="get_odom_robot" output="screen" />  
  </group>

  <!--Get position   
<node pkg="multi_amr_nectec" type="position_robot" name="position_robot" output="screen" >
<param name ="team_size" value = "$(arg team_size)" />
</node>
-->
<group ns="vl_robot">
 <!--Fake odom -->
 <node pkg="multi_amr_nectec" type="fake_odom_vl" name="fake_odom_vl" output="screen">
  <param name ="default_x" value = "0" />
  <param name ="default_y" value = "0" />
</node>

 <!--transform fake_odom to map   -->
 <node pkg="tf" type="static_transform_publisher" name="vl_odom_map_broadcaster" args="0 0 0 0 0 0 /map /odom 20" />
  <!--Get odom from fake robot  -->
<node pkg="multi_amr_nectec" type="get_odom_robot" name="get_odom_robot" output="screen" />
 </group>

  <!--Get position of VL from fake odom  
 <node pkg="multi_amr_nectec" type="position_vl" name="position_vl" output="screen" />
  -->

</launch>